saleor/graphql/page/mutations/attributes.py

```diff
@@ -139,3 +144,91 @@ def perform_mutation(cls, _root, info, **data):
         page_type.page_attributes.remove(*attr_pks)
 
         return cls(page_type=page_type)
+
+
+class PageTypeReorderAttributes(BaseMutation):
+    page_type = graphene.Field(
+        PageType, description="Page type from which attributes are reordered."
+    )
+
+    class Arguments:
+        page_type_id = graphene.Argument(
+            graphene.ID, required=True, description="ID of a page type."
+        )
+        moves = graphene.List(
+            graphene.NonNull(ReorderInput),
+            required=True,
+            description="The list of attribute reordering operations.",
+        )
+
+    class Meta:
+        description = "Reorder the attributes of a page type."
+        permissions = (PageTypePermissions.MANAGE_PAGE_TYPES_AND_ATTRIBUTES,)
+        error_type_class = PageError
+        error_type_field = "page_errors"
+
+    @classmethod
+    def prepare_operations(cls, moves: ReorderInput, page_attributes: "QuerySet"):
+        """Prepare operations dict for reordering attributes.
+
+        Operation dict format:
+            key: page attribute pk,
+            value: sort_order value - relative sorting position of the attribute
+        """
+        operations = {}
+        invalid_attributes = []
+
+        # resolve attribute moves
+        for move_info in moves:
+            attribute_pk = from_global_id_strict_type(
+                move_info.id, only_type=Attribute, field="pk"
+            )
+
+            try:
+                page_attribute = page_attributes.get(attribute_id=int(attribute_pk))
```

Comment:
Maybe instead of getting 1 attribute by one from database, we can fetch ids of page attributes?